### PR TITLE
Update problems.tsx with debouncing

### PR DESCRIPTION
**Pull request by mjh316**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [ x] I have tested my code.
- [ x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [ x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ x] I have linked this PR to any issues that it closes.

### CODE REVIEW

**Comment by SansPapyrus683** [comment]
so what does this do

File: src/pages/problems.tsx [hunk 1]
```diff
@@ -32,6 +33,25 @@ type DataProps = {
...
+};
+
+// function queryHook(query: string, search: (value: string) => void) {
+//   if (timerId) {
+//     clearTimeout(timerId);
+//   }
+
+//   timerId = setTimeout(() => search(query), timeout);
+//}
+
```
Comment (thecodingwizard) [inline src/pages/problems.tsx:54]: this can probably be deleted?

File: src/pages/problems.tsx [hunk 2]
```diff
@@ -122,7 +142,9 @@ export default function ProblemsPage(props: PageProps<DataProps>) {
               <h1 className="text-center text-3xl sm:text-5xl font-bold text-white dark:text-dark-high-emphasis mb-6">
                 Problems (Beta)
               </h1>
-              <SearchBox />
+              <SearchBox
+              // queryHook={queryHook}
+              />
```
Comment (thecodingwizard) [inline src/pages/problems.tsx:147]: similarly here

**Review by thecodingwizard — APPROVED** [review]
lgtm other than removing some outdated code. thanks for implementing this!!

**Comment by mjh316** [comment]
@SansPapyrus683 It just basically makes it so that the /problems page doesn't update every character you type, and instead updates every 200 milliseconds (debouncing!). This is so we don't use up all our credits lmao. (also if it looks good, could you approve)

**Review by thecodingwizard — APPROVED** [review]

**Pull request merged**